### PR TITLE
HTTP Basic Auth support is added for check-es-shard-allocation-status.rb

### PR DESCRIPTION
**Pull request by cihangirbesiktas**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [ ] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass 

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
+- HTTP Basic Auth support is added for shard allocation status (@cihangirbesiktas)
```
Comment (majormoses) [inline CHANGELOG.md]: Add this under a `### Added` header.

File: bin/check-es-shard-allocation-status.rb [hunk 2]
```diff
@@ -58,13 +59,40 @@ class ESShardAllocationStatus < Sensu::Plugin::Check::CLI
...
+    if config[:user] && config[:password]
+      auth = 'Basic ' + Base64.strict_encode64("#{config[:user]}:#{config[:password]}").chomp
+      headers = { 'Authorization' => auth }
+    end
+
+    r = RestClient::Resource.new("#{config[:scheme]}://#{config[:server]}:#{config[:port]}#{resource}", timeout: config[:timeout], headers: headers)
     JSON.parse(r.get)
   rescue Errno::ECONNREFUSED
-    warning 'Connection refused'
+    critical 'Connection refused'
```
Comment (majormoses) [inline bin/check-es-shard-allocation-status.rb]: This makes this a breaking change, let's leave this alone and open a separate issue to discuss this change.

**Comment by cihangirbesiktas** [comment]
I do not see any unit test on the plugin, but the changes are working on our cluster. If you want further information, we can attach them.

**Comment by majormoses** [comment]
@cihangirbesiktas yes unfortunately there is a lack of testing across most of the plugins which is something we are looking to change. We are currently in the process of formalizing new requirements for PRs which is why I am asking for a testing artifact and not requiring unit tests yet. If you can show me a gist, screenshot, log message, etc of redacted input and output that will be good enough for me.

**Comment by majormoses** [comment]
The code looks good but we are trying to avoid requiring maintainers to always pull down and do their own testing (sometimes it's not even an option as it requires a paid service) as this is unsustainable with the current active number of maintainers.

**Comment by cihangirbesiktas** [comment]
@majormoses I attached the screenshot of the check which is run in a docker sensu client.
es-shard-allocation

**Comment by majormoses** [comment]
thanks, I will merge and release this shortly.

**Comment by majormoses** [comment]
only thing I see missing which I can add on merge is that you also added a timeout option that was not mentioned in your changelog entry.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-elasticsearch/versions/1.4.0

**Pull request merged**